pkg/kubelet/cm/dra/types.go

```diff
@@ -26,9 +26,12 @@ type ActivePodsFunc func() []*v1.Pod
 
 // Manager manages all the DRA resource plugins running on a node.
 type Manager interface {
-	// Allocate prepares and assigns resources to a container in a pod. From
-	// the requested resources, Allocate will communicate with the
-	// DRA resource plugin to prepare resources.
+	// PrepareResources prepares resources for a container in a pod.
+	// It communicates with the DRA resource plugin to prepare resources and
+	// returns container annotations to trigger CDI injection.
+	PrepareResources(pod *v1.Pod, container *v1.Container) ([]kubecontainer.Annotation, error)
+
+	// Allocate method is defined to satisfy HintProvider interface
 	Allocate(pod *v1.Pod, container *v1.Container) error
```

Comment:
If we are planning to move this out of the pod admission loop, we can drop this.

Reply:
Thank you for the review!

Yes, I'm going to get rid of Topology-related code in the next PR.

Reply:
done, see "kubelet: get rid of DRA Topology-related code" commit in this PR for more details.